packages/app/src/app/overmind/namespaces/editor/state.ts

```diff
@@ -157,13 +165,34 @@ export const state: State = {
       );
 
       // If this is a single directory with no children
-      if (!Object.keys(modulesObject).some(p => dirname(p) === path)) {
-        modulesObject[path] = { ...d, type: 'directory' };
+      if (!Object.keys(paths).some(p => dirname(p) === path)) {
+        paths[path] = { shortid: d.shortid, type: 'directory' };
       }
     });
 
-    return modulesObject;
+    return paths;
   },
+  modulesByPath: ({ currentSandbox, modulePaths }) =>
+    Object.keys(modulePaths).reduce((aggr, path) => {
```

Comment:
We can make this a bit faster by having something like this:

```
const finalObj = {};

Object.keys(modulePaths).forEach(m => {
  finalObj[...] = m;
});

return finalObj;
```

This way we don't create a new object and merge for all dirs and modules (so max ~300 times) and should be a bit faster.

Reply:
ah, but the populated `type` property of "file" or "directory" I believe to be used inside BrowserFS?

Reply:
Yes, oh, this is just pseudo code. What comes after the `=` should stay the same.

Reply:
Hm... then I do not see how they differ? :) You create a new object and populate it in both scenarios? The reducer is not returning a new object every time, it is returning the same object, just populated with a new property :)

Reply:
But reducers... they do not express populating arrays/objects the best. Your way is more explicit for sure. I can change it to make it more readable :)

Reply:
Ahhh, I've been misreading this. Sorry! I thought it was saying

```
return {
	...acc,
	[m.path]: {...}
}
```

Whoops sorry!